integration/apidefinition_test.go

```diff
@@ -508,3 +515,122 @@ func TestApiDefinitionCreateIgnored(t *testing.T) {
+		dashboardLocalHost  = "http://localhost:7200"
+	)
+
+	if strings.TrimSpace(os.Getenv("TYK_MODE")) == "ce" {
+		t.Log("CE is not feasible to test at the moment.")
+		return
+	}
+
+	adCreate := features.New("Create an ApiDefinition for Upstream TLS").
+		Setup(func(ctx context.Context, t *testing.T, envConf *envconf.Config) context.Context {
+			testNS := ctx.Value(ctxNSKey).(string) //nolint: errcheck
+			is := is.New(t)
+			t.Log(testNS)
+			_, err := createTestTlsSecret(ctx, testNS, nil, envConf)
+			is.NoErr(err)
+
+			return ctx
+		}).
+		Setup(func(ctx context.Context, t *testing.T, envConf *envconf.Config) context.Context {
+			testNS := ctx.Value(ctxNSKey).(string) //nolint:errcheck
+			is := is.New(t)
+
+			// Create ApiDefinition with Upstream certificate
+			_, err := createTestAPIDef(ctx, testNS, func(apiDef *v1alpha1.ApiDefinition) {
+				apiDef.Name = apiDefUpstreamCerts
+				apiDef.Spec.UpstreamCertificateRefs = map[string]string{
+					"*": certName,
+				}
+				apiDef.Spec.VersionData.DefaultVersion = defaultVersion
+				apiDef.Spec.VersionData.NotVersioned = true
+				apiDef.Spec.VersionData.Versions = map[string]model.VersionInfo{
+					defaultVersion: {Name: defaultVersion},
+				}
+			}, envConf)
+			is.NoErr(err) // failed to create apiDefinition
+
+			return ctx
+		}).
+		Assess("ApiDefinition must have upstream field defined",
+			func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+				is := is.New(t)
+				client := cfg.Client()
+				testNS := ctx.Value(ctxNSKey).(string) //nolint:errcheck
+
+				tlsSecret := v1.Secret{} //nolint:errcheck
+
+				err := client.Resources(testNS).Get(ctx, certName, testNS, &tlsSecret)
+				is.NoErr(err)
+
+				certPemBytes, ok := tlsSecret.Data["tls.crt"]
+				if !ok {
+					is.Fail()
+				}
+				certFingerPrint := cert.CalculateFingerPrint(certPemBytes)
+
+				opConfSecret := v1.Secret{}
+				err = client.Resources(opNs).Get(ctx, "tyk-operator-conf", opNs, &opConfSecret)
+				is.NoErr(err)
+
+				tykAuth, ok := opConfSecret.Data["TYK_AUTH"]
+				if !ok {
+					is.Fail()
+				}
+
+				tykOrg, ok := opConfSecret.Data["TYK_ORG"]
+				if !ok {
+					is.Fail()
+				}
+
+				calculatedCertID := string(tykOrg) + certFingerPrint
+
+				err = wait.For(func() (done bool, err error) {
+					hc := &http.Client{}
+
+					req, err := http.NewRequest(
+						http.MethodGet,
+						fmt.Sprintf("%s/api/certs/?certId=%s&org_id=%s", dashboardLocalHost, calculatedCertID, string(tykOrg)),
+						nil,
+					)
+					is.NoErr(err)
+					req.Header.Add("Content-type", "application/json")
+					req.Header.Add("authorization", string(tykAuth))
+
+					resp, err := hc.Do(req)
+					is.NoErr(err)
+
+					response, err := io.ReadAll(resp.Body)
+					if err != nil {
+						return false, nil
+					}
+
+					certResponse := struct {
+						Certs []string `json:"certs"`
+						Pages int      `json:"pages"`
+					}{}
+					err = json.Unmarshal(response, &certResponse)
```

Comment:
We can inline this error check

Reply:
I believe this way of handling errors is much easier to read and adds no additional benefit.